scripts/install-packages.R

```diff
@@ -9,7 +9,26 @@ if ("pacman" %in% rownames(installed.packages()) == FALSE) install.packages("pac
 # OG package list: pacman::p_load(git2r, httr, semver, testthat, yaml)
 
 pacman::p_load(dplyr, ggplot2)
+pacman::p_load_gh("jimjunker1/junkR")
 
-trend_detect <- function(){
+#' This function calculates the trend and assigns the study to a category: "trend"
+#' "trend" is a categorical variable with three levels: "positive", "negative", and "no change"
+#' Trend is calculated based on OLS regression 
+#' 
+#' @param data: A data.frame or data.frame confomrable object to perform analysis. Must include 'Year' and 'Number' columns at minimum.
+#' @param cols_keep: A string of the columns to keep in the analysis. This defaults to NULL, but will always include 'Year' and 'Number'
+#' @param 
+trend_detect <- function(data, cols_keep = NULL,...){
+  if(!is.data.frame(data)) data <- data.frame(data)
   
-}
\ No newline at end of file
+  if(is.null(cols_keep)){data_mod = data[c("Year","Number")]} else{data_mod = data[c(cols_keep, "Year","Number")]}
+  trend_summ <- summary(lm(Number~Year, data = data_mod))$coefficients
+  df_add <- data.frame(trend = NA, coef = trend_summ['Year','Estimate'], coef_se = trend_summ['Year','Std. Error'])
+  if(trend_summ['Year','Pr(>|t|)'] >= 0.05) {
+    df_add[,'trend'] <- "no change"
+  } else if(trend_summ['Year','Estimate'] > 0) {
+    df_add[,'trend'] <- "positive"
+  } else {df_add[,'trend'] <- "negative"}
+  if(is.null(cols_keep)){return(df_add)} else{cbind(unique(data_mod[,names(data_mod) %in% cols_keep]), df_add)}
+}
+  
```

Comment:
This all looks good Junker. I spot checked 5 studies to see if the coefs were correct and they looked good. Sweet work. Did you write your own function to name objects in your list after you split by grouping variables? If so you should send it over to me because I've needed something like that!